cxx-squid/src/main/java/org/sonar/cxx/CxxVCppBuildLogParser.java

```diff
@@ -324,155 +351,101 @@ private void ParseCommonCompilerOptions(String line, String fileElement) {
       AddMacro("_WCHAR_T_DEFINED", fileElement);   
       AddMacro("_NATIVE_WCHAR_T_DEFINED", fileElement);
     }      
-    
-    //_Wp64 Defined when specifying /Wp64.
+
+    //_Wp64 Defined when specifying /Wp64. Deprecated in Visual Studio 2010 and Visual Studio 2012, and not supported starting in Visual Studio 2013
     if (line.contains("/Wp64 ")) {
       AddMacro("_Wp64", fileElement);    
     }   
-    
+
     //_M_AMD64 Defined for x64 processors.
     //_WIN32 Defined for applications for Win32 and Win64. Always defined.
     //_WIN64 Defined for applications for Win64.
     //_M_X64 Defined for x64 processors.    
     //_M_IX86 Defined for x86 processors. See the Values for _M_IX86 table below for more information. This is not defined for x64 processors.
     //_M_IA64 Defined for Itanium Processor Family 64-bit processors.
-    //_M_IX86_FP Expands to a value indicating which /arch compiler option was used:
-    //    0 if /arch was not used.    
-    //    1 if /arch:SSE was used.
-    //    2 if /arch:SSE2 was used.   
-    if(platform.equals("x64")) {
+    if(platform.equals("x64")||line.contains("/D WIN64")) {
+      // Defined for compilations that target x64 processors.
       AddMacro("_WIN32", fileElement);
+      // This is not defined for x86 processors.
       AddMacro("_WIN64", fileElement);
       AddMacro("_M_X64", fileElement);
       AddMacro("_M_IA64", fileElement);
       AddMacro("_M_AMD64", fileElement);
     }
-    
-    if(platform.equals("Win32")) {
+    else if(platform.equals("Win32")||line.contains("/D WIN32")) {

```

Comment:
i think this is problematic, if you have a configuration where you build both win32 and x64 you will be defining all macros to all files. because all of them apply to all files i would still not do this. instead i would keep it as it was so that only one set of macros is defined for the all source code. we have a few of these were we build everything in x64 and because of single project that builds in win32 it will cause issues.

we can do this when we set the macros, defines per file